punxa/fpu.py

```diff
@@ -1273,7 +1273,7 @@ def convert_hp_to_i32(self, v):
             if (a.compare(b) != 0):
                 self.cpu.csr[CSR_FFLAGS] = CSR_FFLAGS_INEXACT_MASK
                 
-        return signExtend(r, 32, 64) 
+        return r#signExtend(r, 32, 64)
```

Comment:
fpu is shared with 64 bit version of the processor, are you shure your change to convert_hp_to_i64 is not breaking the 64 bits version ?
Why RV32 processor should be using convert_hp_to_i64 ?

Reply:
Resolved. There was an incompatibility with the functions: convert_sp_to_i32, convert_hp_to_u32, and convert_hp_to_i32. I have duplicated the functions to create separate versions for 64-bit and 32-bit systems.